packages/select/src/vaadin-select-base-mixin.js

```diff
@@ -0,0 +1,622 @@
+/**
+ * @license
+ * Copyright (c) 2017 - 2023 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { setAriaIDReference } from '@vaadin/a11y-base/src/aria-id-reference.js';
+import { DelegateFocusMixin } from '@vaadin/a11y-base/src/delegate-focus-mixin.js';
+import { KeyboardMixin } from '@vaadin/a11y-base/src/keyboard-mixin.js';
+import { DelegateStateMixin } from '@vaadin/component-base/src/delegate-state-mixin.js';
+import { MediaQueryController } from '@vaadin/component-base/src/media-query-controller.js';
+import { OverlayClassMixin } from '@vaadin/component-base/src/overlay-class-mixin.js';
+import { TooltipController } from '@vaadin/component-base/src/tooltip-controller.js';
+import { generateUniqueId } from '@vaadin/component-base/src/unique-id-utils.js';
+import { FieldMixin } from '@vaadin/field-base/src/field-mixin.js';
+import { LabelController } from '@vaadin/field-base/src/label-controller.js';
+import { ButtonController } from './button-controller.js';
+
+/**
+ * @polymerMixin
+ * @mixes DelegateFocusMixin
+ * @mixes DelegateStateMixin
+ * @mixes FieldMixin
+ * @mixes KeyboardMixin
+ * @mixes OverlayClassMixin
+ */
+export const SelectBaseMixin = (superClass) =>
+  class SelectBaseMixin extends OverlayClassMixin(
+    DelegateFocusMixin(DelegateStateMixin(KeyboardMixin(FieldMixin(superClass)))),
```

Comment:
I'm a bit confused about `ControllerMixin`, which is pulled in here implicitly by `FieldMixin` and how this affects `LitElement` based components.

For `vaadin-avatar` I explicitly excluded `ControllerMixin` from the Lit-based element, because I adopted the approach from `vaadin-button`. `vaadin-checkbox` on the other hand includes `ControllerMixin` in both versions of elements. There's also some duplicate code on all of these to register the tooltip controller, which I assumed was due to `ControllerMixin`.

So I'm wondering if:
- there can be any side-effects when using this with `LitElement`, which has its own controller implementation
- we can remove the duplicate controller registration, for tooltip controller for example, from existing elements
- we should make `ControllerMixin` only apply if `superClass` is not based on `LitElement`

Reply:
Good point. We can probably check for `typeof addController === 'function'`. I will create a PR.

Reply:
@sissbruecker Created #6197 to handle this.